### DISCUSSION TITLE
How to output the f_avg, f_min, cv_min, cv_avg printed when verbose=True?

### DISCUSSION THREAD

**Discussion by zhilei9** [anyoptimization/pymoo#659] [Q&A]
After running the GA optimization with constraints, I printed the following information. Could anyone teach me how to output the f_avg, f_min, cv_min, cv_avg to a csv file. 

=================================================================================
n_gen  |  n_eval  |     cv_min    |     cv_avg    |     f_avg     |     f_min    
=================================================================================
     1 |      100 |  0.000000E+00 |  0.2536728846 |  1.104550E+07 | -7.762836E+07

I tried the following method. However, the output values are not exactly the same as those in the above table:
    optimization_info = pd.DataFrame({
        'n_gen': [gen + 1 for gen in range(res.algorithm.n_gen)] [:-1],
        'favg': [np.mean(-h.pop.get("F")) for h in res.history]
    })
    optimization_info.to_csv(config.save_path / 'optimization_info.csv', index=False)

**Comment by blankjul** [comment]
`f` refers to the objective space and `cv` to the constraint space.
`min` is the minimum and `avg` the average.
So the value,
`cv_min` refers to the minimum constraint violation value in the current population.

Analogously, you can derive the meaning for the other variables.
